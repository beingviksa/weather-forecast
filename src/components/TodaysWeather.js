const TodaysWeather = ({ city, minMaxTemp }) => {
  const { name, country } = city;
  return (
    <div className="todays-forecast">
      <h1 data-testid="city-country">
        {name} {country ? `(${country})` : ""}
      </h1>
      <section>
        <div className="temp-details">
          <span className="temp-label">Min.</span>
          <span className="min-temp" data-testid="min-temp">
            {minMaxTemp.min}&deg;C
          </span>
        </div>
        <div className="temp-details">
          <span className="temp-label">Max.</span>
          <span className="max-temp" data-testid="max-temp">
            {minMaxTemp.max}&deg;C
          </span>
        </div>
      </section>
    </div>
  );
};

export default TodaysWeather;
