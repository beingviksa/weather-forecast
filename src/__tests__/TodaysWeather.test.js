import { render } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import TodaysWeather from "../components/TodaysWeather";

describe("<TodaysWeather/>", () => {
  const props = {
    city: {
      id: 2643743,
      name: "London",
      state: "",
      country: "GB",
      coord: {
        lon: -0.12574,
        lat: 51.50853,
      },
    },
    minMaxTemp: {
      min: "5",
      max: "17",
    },
  };
  it("render the TodaysWeather Component", async () => {
    const { container } = render(
      <Router>
        <TodaysWeather {...props} />
      </Router>
    );
    expect(container).toMatchSnapshot();
  });

  it("render the city name with country", async () => {
    const { getByTestId } = render(
      <Router>
        <TodaysWeather {...props} />
      </Router>
    );

    const placeName = getByTestId("city-country");
    expect(placeName.textContent).toBe("London (GB)");
  });

  it("render the min and max temperature", async () => {
    const { getByTestId } = render(
      <Router>
        <TodaysWeather {...props} />
      </Router>
    );

    const minTemp = getByTestId("min-temp");
    expect(minTemp.textContent).toBe("5°C");

    const maxTemp = getByTestId("max-temp");
    expect(maxTemp.textContent).toBe("17°C");
  });
});
