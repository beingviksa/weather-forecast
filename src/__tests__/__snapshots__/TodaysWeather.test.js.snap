// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodaysWeather/> render the TodaysWeather Component 1`] = `
<div>
  <div
    class="todays-forecast"
  >
    <h1
      data-testid="city-country"
    >
      London
       
      (GB)
    </h1>
    <section>
      <div
        class="temp-details"
      >
        <span
          class="temp-label"
        >
          Min.
        </span>
        <span
          class="min-temp"
          data-testid="min-temp"
        >
          5
          °C
        </span>
      </div>
      <div
        class="temp-details"
      >
        <span
          class="temp-label"
        >
          Max.
        </span>
        <span
          class="max-temp"
          data-testid="max-temp"
        >
          17
          °C
        </span>
      </div>
    </section>
  </div>
</div>
`;
